samples/server/go-server/api/server/storage/pgsqlstore.go

```diff
@@ -126,22 +130,36 @@ func (pg *pgSQLStore) GetProject(pID string) (*pb.Project, error) {
 	return &pb.Project{Name: pName}, nil
 }
 
-// ListProjects returns the project id for all projects from the store
-func (pg *pgSQLStore) ListProjects(filters string) ([]*pb.Project, error) {
-	rows, err := pg.DB.Query(listProjects)
+var paginationKey = "XxoPtCUzrUv4JV5dS+yQ+MdW7yLEJnRMwigVY/bpgtQ="
+
+// ListProjects returns up to pageSize number of projects beginning at pageToken (or from
+// start if pageToken is the emtpy string).
+func (pg *pgSQLStore) ListProjects(filter string, pageSize int, pageToken string) ([]*pb.Project, string, error) {
+	var rows *sql.Rows
+	id, err := decryptInt64(pageToken, paginationKey)
+	if err == nil {
+		rows, err = pg.DB.Query(listProjectsFromPage, pageSize, id)
+	} else {
+		rows, err = pg.DB.Query(listProjects, pageSize)
+	}
 	if err != nil {
-		return nil, status.Error(codes.Unknown, fmt.Sprintf("Failed to list Projects from database"))
+		return nil, "", status.Error(codes.Unknown, "Failed to list Projects from database")
```

Comment:
codes.Internal is probably better here and below

Reply:
Agree. Fixed in 2c060cf